src/Data/Array/ST.ss

```diff
@@ -0,0 +1,110 @@
+  (export new
+          unshiftAllImpl
+          spliceImpl
+          pokeImpl
+          peekImpl
+          popImpl
+          unsafeThawImpl
+          unsafeFreezeImpl
+          lengthImpl
+          shiftImpl
+          pushImpl
+          pushAllImpl
+          toAssocArrayImpl
+          sortByImpl
+          freeImpl
+          thawImpl
+          freezeImpl)
+  (import (only (rnrs base) define lambda error list let begin cons if and < > >= +)
+          (only (chezscheme) display format newline)
+          (only (rnrs sorting) list-sort)
+          (prefix (purs runtime lib) rt:)
+          (prefix (purs runtime srfi :214) srfi:214:))
+          
+  (define new
+    (lambda () (rt:make-array)))
+
+  (define lengthImpl
+    (lambda (xs) (rt:array-length xs)))
+
+  (define pokeImpl
+    (lambda (i a xs)
+      (let ([ret (and (>= i 0) (< i (rt:array-length xs)))])
+        (if ret (srfi:214:flexvector-set! xs i a))
+        ret)))
+
+  (define peekImpl
+    (lambda (just nothing i xs)
+      (if (and (>= i 0) (< i (rt:array-length xs)))
+        (just (rt:array-ref xs i))
+        nothing)))
+
+  (define popImpl
+    (lambda (just nothing xs)
+      (if (> (rt:array-length xs) 0)
+        (just (srfi:214:flexvector-remove-back! xs))
+        nothing)))
+
+  (define unshiftAllImpl
+    (lambda (as xs)
+      (srfi:214:flexvector-add-all! xs 0 (srfi:214:flexvector->list as))
+      (rt:array-length xs)))
+
+  (define shiftImpl
+    (lambda (just nothing xs)
+      (if (> (rt:array-length xs) 0)
+        (just (srfi:214:flexvector-remove-front! xs))
+        nothing)))
+
+  (define spliceImpl
+    (lambda (i howMany bs xs)
+      (if (> howMany 0)
+        (let ([removed (srfi:214:make-flexvector howMany)])
+          (srfi:214:flexvector-copy! removed 0 xs i (+ i howMany))
+          (srfi:214:flexvector-remove-range! xs i (+ i howMany))
+          (srfi:214:flexvector-add-all! xs i (srfi:214:flexvector->list bs))
+          removed)
+        (begin
+          (srfi:214:flexvector-add-all! xs i (srfi:214:flexvector->list bs))
+          (rt:make-array)))))
+
+  (define unsafeThawImpl
+    (lambda (xs) xs))
+
+  (define unsafeFreezeImpl
+    (lambda (xs) xs))
+
+  (define pushImpl
+    (lambda (a xs)
+      (srfi:214:flexvector-add-back! xs a)
+      (rt:array-length xs)))
+
+  (define pushAllImpl
+    (lambda (as xs)
+      (srfi:214:flexvector-append! xs as)
+      (rt:array-length xs)))
+
+  (define sortByImpl
+    (lambda (compare fromOrdering xs)
+      (srfi:214:list->flexvector
+        (list-sort
+          (lambda (x y) (> (fromOrdering ((compare y) x)) 0))
+          (srfi:214:flexvector->list xs)))))
+
+  (define toAssocArrayImpl
+    (lambda (xs)
+      (srfi:214:flexvector-map/index
+        (lambda (i x)
+          (rt:make-object (cons "index" i) (cons "value" x)))
+        xs)))
+            
+  (define copyImpl
+    (lambda (xs) (srfi:214:flexvector-copy xs)))
+
+  (define freeImpl copyImpl)
```

Comment:
Typo? This binding isn't defined in JS/PS files.

Reply:
Ah yes, a weird typo, maybe from "freeze".